### DISCUSSION TITLE
关于用户词典userdb自定义修改问题

### DISCUSSION THREAD

**Discussion by Randonyld** [rime/home#878] [Q&A]
通过同步或者导出生成的userdb文件，记录了用户的历史输入习惯。但是在平时使用过程中经常会出现打错字或选错词的问题，导致这些错误信息均被记录在档案中，在迁移的时候想剔除这些错误。
userdb文件中每个此条均有c d t三个参数，c 应该是该词条输入的次数，d 类似于词条频率，t 好像是所有词条的总输入次数（t=sum(c_i)? 但有时并不相等，利用shift+delete删除词条会导致该词条c=-1，并不会在档案中删除该词条）
本人不是专业开发者，很难从librime源文件中找到userdb生成规则的相关代码。请问各位大佬如何更加方便快捷地自定义修正userdb中的记录，而与源代码中的词频生成规则不冲突。

**Comment by lotem** [comment]
手動刪除用戶詞，標記`d`爲負數。
涉及用戶詞典同步時，其他備份裏可能包含刪除前保存的該詞條的記錄，因此刪除的記錄需要保留，才能避免本地刪除後又從其他備份同步回來。
標記爲刪除後不會再顯示爲後選詞，可以放心忽略他的存在。

**Reply by Randonyld** [reply]
了解保留词条的用意了，感谢作者回复！

**Reply by Chinglissian** [reply]
我手動將 userdb.txt 中的 d 值設置爲 -1，但在同步用戶數據後 d 的值又會被覆蓋成 0，哪怕我刪除了對應的 userdb 文件夾再去部署和同步用戶數據也無濟於事，我試圖刪除的選詞在輸入時依舊會出現，請問應該如何讓某一選詞能標記爲永久刪除呢？我不涉及到跨設備和多用戶的情況

**Reply by lotem** [reply]
> 我手動將 userdb.txt 中的 d 值設置爲 -1

不支持手動修改。
請同步後，在選詞界面刪除。

**Reply by Chinglissian** [reply]
> > 我手動將 userdb.txt 中的 d 值設置爲 -1
> 
> 不支持手動修改。 請同步後，在選詞界面刪除。

我是通過 Python 代碼對 userdb.txt 中選詞的值進行修改的，數量比較多。如果我要批量刪除的話，是不是只能改方案名後同步到新方案中了？就不能徹底清空某個方案的用戶數據後再通過 usersb.txt 重新導入嗎？（之前試過批量添加選詞，是沒問題的，結果添加了現在不能批量刪除了 Lol）

**Reply by lotem** [reply]
用戶詞典設計來記錄用戶輸入上屏的字詞。時間信息也很重要。腳本很難模擬用戶的操作。

真正的用戶詞典已經破壞，沒啥可留戀的，那就刪庫重建吧。

要批量加詞最好還是加到 `*.dict.yaml` 文件，比較高效。
